core/src/main/java/org/stellar/anchor/asset/AssetServiceValidator.java

```diff
@@ -5,43 +5,65 @@
 import java.util.HashSet;
 import java.util.Set;
 import lombok.RequiredArgsConstructor;
+import org.stellar.anchor.api.asset.AssetInfo;
+import org.stellar.anchor.api.asset.FiatAssetInfo;
+import org.stellar.anchor.api.asset.StellarAssetInfo;
 import org.stellar.anchor.api.exception.InvalidConfigException;
-import org.stellar.anchor.api.sep.AssetInfo;
 
 @RequiredArgsConstructor
 public class AssetServiceValidator {
 
   public static void validate(AssetService assetService) throws InvalidConfigException {
     // Check for non-zero assets
-    if (assetService == null || isEmpty(assetService.listAllAssets())) {
+    if (assetService == null || isEmpty(assetService.getAllAssets())) {
       throw new InvalidConfigException("0 assets defined in configuration");
     }
 
     // Check for duplicate assets
     Set<String> existingAssetNames = new HashSet<>();
-    for (AssetInfo asset : assetService.listAllAssets()) {
-      if (asset != null && !existingAssetNames.add(asset.getSep38AssetName())) {
+    for (AssetInfo asset : assetService.getAllAssets()) {
+      if (asset != null && !existingAssetNames.add(asset.getId())) {
         throw new InvalidConfigException(
-            "Duplicate assets defined in configuration. Asset = " + asset.getSep38AssetName());
+            "Duplicate assets defined in configuration. Asset = " + asset.getId());
       }
     }
 
-    // Asset level validation
-    for (AssetInfo assetInfo : assetService.listAllAssets()) {
+    // Validate stellar asset
+    for (StellarAssetInfo assetInfo : assetService.getStellarAssets()) {
       validateWithdraw(assetInfo);
       validateDeposit(assetInfo);
+      validateQuoteSupportedIfRequired(assetInfo);
       // TODO: ANCHOR-374 add more validation
     }
+
+    for (FiatAssetInfo assetInfo : assetService.getFiatAssets()) {
+      validateQuoteSupportedIfRequired(assetInfo);
+      // TODO: ANCHOR-374 add more validation
+    }
+  }
+
+  private static void validateQuoteSupportedIfRequired(AssetInfo assetInfo)
+      throws InvalidConfigException {
+    // Validate SEP-31 fields
+    if (assetInfo.getSep31() != null && assetInfo.getSep31().getEnabled()) {
+      boolean isQuotesSupported = assetInfo.getSep31().isQuotesSupported();
+      boolean isQuotesRequired = assetInfo.getSep31().isQuotesRequired();
+      if (isQuotesRequired && !isQuotesSupported) {
+        throw new InvalidConfigException(
+            "if quotes_required is true, quotes_supported must also be true for asset: "
```

Comment:
Hmm... do we need `quotes_supported` then?

Reply:
This is also part of the https://stellarorg.atlassian.net/browse/ANCHOR-794 we will remove once confirm its not needed. This PR does not make modification to any existing fields/function/validation etc..